pkg/utils/retry/retry.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package retry
+
+import (
+	"time"
+)
+
+// Params are used to define how retry attempts are handled.
+type Params struct {
+	MaxRetries     uint
+	InitialBackoff uint
+	BackoffFactor  float64
+}
+
+// Invocation represents a function that is desired to be retried until some condition is met. It returns:
+// bool: Indicates whether the retry condition is met. False = more retries desired. True = no more retries desired.
+// interface{}: This is the returned data that the caller wants. The caller is expected to know the underlying type.
+// error: Used to indicate whether an error occurred.
+type Invocation func() (bool, interface{}, error)
+
+// Retry retries the given Invocation based on the given Params until the Invocation indicates that it's been
+// successful. If successful, this function immediately returns the value and error returned from the given Invocation.
+// If retry attempts are exhausted, this function returns the most recent value and error returned from the
+// given Invocation.
+func Retry(funcToRetry Invocation, params *Params) (interface{}, error) {
+	var value interface{}
+
+	var err error
+
+	var backoff time.Duration
+
+	if params.MaxRetries != 0 {
+		// InitialBackoff in the retry parameters is in milliseconds, but a time.Duration is an int
```

Comment:
You don't need to do this conversion if InitialBackoff was time.Duration.

Reply:
Done.